pkg/logs/sender/sender.go

```diff
@@ -53,8 +54,14 @@ func (s *Sender) run() {
 // and try to send the message to the additional destinations only once.
 func (s *Sender) send(payload message.Message) {
 	for {
-		err := s.destinations.Main.Send(payload) // this call is blocking until the inner connection is properly established
+		// this call is blocking until payload is sent (or the connection destination context cancelled)
+		err := s.destinations.Main.Send(payload)
 		if err != nil {
+			if err == context.Canceled {
```

Comment:
Wondering whether we could merge this check with the switch right bellow.

Reply:
did not find how to do that :/